app/services/amr/validate_and_persist_readings_service.rb

```diff
@@ -0,0 +1,17 @@
+module Amr
+  class ValidateAndPersistReadingsService
+    def initialize(school)
+      @school = school
+      @meter_collection = AmrMeterCollection.new(@school)
+    end
+
+    def perform
+      return unless @school.meters_with_readings.any?
+      p "Validate and persist readings for #{@school.name} #{@school.id}"
+      AggregateDataService.new(@meter_collection).validate_meter_data
+      UpsertValidatedReadings.new(@meter_collection).perform
```

Comment:
This does something in place and then the service returns it - which references @jamesalmond above